docs/src/main/asciidoc/security-built-in-authentication-support-concept.adoc

```diff
@@ -187,7 +187,40 @@ public class HelloService {
 
 === How to customize authentication exception responses
 
-By default, the authentication security constraints are enforced before the JAX-RS chain starts and only way to handle Quarkus Security authentication exceptions is to provide a failure handler like this one:
+You can use JAX-RS `ExceptionMapper` to capture Quarkus Security authentication exceptions such as `io.quarkus.security.AuthenticationFailedException`, for example:
+
+[source,java]
+----
+package io.quarkus.it.keycloak;
+
+import javax.annotation.Priority;
+import javax.ws.rs.Priorities;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
+
+import io.quarkus.security.AuthenticationFailedException;
+
+@Provider
+@Priority(Priorities.AUTHENTICATION)
+public class AuthenticationFailedExceptionMapper implements ExceptionMapper<AuthenticationFailedException> {
+
+    @Context
+    UriInfo uriInfo;
+
+    @Override
+    public Response toResponse(AuthenticationFailedException exception) {
+        return Response.status(401).header("WWW-Authenticate", "Basic realm=\"Quarkus\"").build();
+    }
+}
+----
+
+CAUTION: By default, authentication mechanisms like `io.quarkus.oidc.runtime.OidcAuthenticationMechanism` needs to handle Quarkus Security authentication exceptions themselves in order redirect request and set correct response status.
```

Comment:
```suggestion
CAUTION: Some HTTP authentication mechanisms such as `io.quarkus.oidc.runtime.OidcAuthenticationMechanism` need to handle authentication exceptions themselves in order to create a correct redirect URL, status code, cookies, etc.
```

Reply:
Sorry, just realized OIDC handles bearer tokens as well which is just 401, so we need to be specific it is the code flow exceptions which should not be customized